modules/mono/csharp_script.cpp

```diff
@@ -437,6 +437,10 @@ static String variant_type_to_managed_name(const String &p_var_type_name) {
 		return "Collections.Dictionary";
 	}
 
+	if (p_var_type_name.begins_with(Variant::get_type_name(Variant::ARRAY) + "[")) {
+		return "Collections.Array<" + p_var_type_name.trim_prefix(Variant::get_type_name(Variant::ARRAY) + "[").trim_suffix("]") + ">";
```

Comment:
This assumes that for `Array[T]`, the `T` string is the same in C#. But that's not necessarily true, for example: `int` is `long`, `AABB` is `Aabb`, etc. Take a look at the `var_types` variable below that contains the mapping.

Also, I don't understand why this change is needed. I thought this was a GDScript PR. C# doesn't support generated documentation yet.